modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -51,16 +51,16 @@ def wait_computations_if_benchmark_mode(func):
     -----
     `func` should return NumPy array with partitions.
     """
-    if BenchmarkMode.get():
-
-        @wraps(func)
-        def wait(cls, *args, **kwargs):
-            """Wait for computation results."""
-            result = func(cls, *args, **kwargs)
-            if isinstance(result, tuple):
-                partitions = result[0]
-            else:
-                partitions = result
+
+    @wraps(func)
```

Comment:
Ideally this should be done in a subscribe way - the whole logic of configuration objects was constructed to allow subscribing to changes and changing the code behaviour only once per configuration change (in this case when benchmark mode goes from on to off or vise versa).

That said, I don't have a good suggestion on how to change this particular piece...